omnibus/files/private-chef-ctl-commands/cleanup_bifrost.rb

```diff
@@ -0,0 +1,235 @@
+require 'pg'
+
+CREATE_SQL =<<SQL
+BEGIN;
+CREATE UNLOGGED TABLE IF NOT EXISTS cleanup_tracking_auth_actors(
+    authz_id  CHAR(32)
+);
+
+CREATE OR REPLACE RULE cleanup_auth_actor_creation_tracking AS ON INSERT TO auth_actor DO INSERT INTO cleanup_tracking_auth_actors VALUES (NEW.authz_id);
+COMMIT;
+SQL
+
+CREATE_SQL2 =<<SQL
+CREATE UNLOGGED TABLE IF NOT EXISTS cleanup_known_auth_actors(
+    authz_id  CHAR(32)
+);
+SQL
+
+DELETE_SQL =<<SQL
+BEGIN;
+DROP RULE IF EXISTS cleanup_auth_actor_creation_tracking ON auth_actor;
+DROP TABLE IF EXISTS cleanup_tracking_auth_actors;
+DROP TABLE IF EXISTS cleanup_known_auth_actors;
+COMMIT;
+SQL
+
+CLEANUP_SQL = <<SQL
+WITH good_auth_actors AS (
+         SELECT authz_id
+         FROM cleanup_tracking_auth_actors
+              UNION
```

Comment:
Minor indent glitch here? Nice use of WITH here; it really helps readability for these kinds of multiclause queries.